modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/io/RowVersionDataIo.java

```diff
@@ -50,46 +52,102 @@ protected RowVersionDataIo(int ver) {
     }
 
     @Override
-    protected void writeRowData(long pageAddr, int dataOff, int payloadSize, RowVersion row, boolean newRow) {
+    protected void writeRowData(long pageAddr, int dataOff, int payloadSize, RowVersion rowVersion, boolean newRow) {
         assertPageType(pageAddr);
 
         long addr = pageAddr + dataOff;
 
         putShort(addr, 0, (short) payloadSize);
         addr += Short.BYTES;
 
-        addr += HybridTimestamps.writeTimestampToMemory(addr, 0, row.timestamp());
+        addr += HybridTimestamps.writeTimestampToMemory(addr, 0, rowVersion.timestamp());
 
-        addr += writePartitionless(addr, row.nextLink());
+        addr += writePartitionless(addr, rowVersion.nextLink());
 
-        putInt(addr, 0, row.valueSize());
+        putInt(addr, 0, rowVersion.valueSize());
         addr += Integer.BYTES;
 
-        putByteBuffer(addr, 0, row.value());
+        BinaryRow row = rowVersion.value();
+
+        if (row != null) {
+            // Write schema version in LE order.
+            putByte(addr, 0, (byte) row.schemaVersion());
+            putByte(addr, 1, (byte) (row.schemaVersion() >> Byte.SIZE));
```

Comment:
What's wrong with `putShort`? We use native ordering here, that's fine

Reply:
That's not fine, because we wrap bytes that we read from the storage into a `ByteBufferRow` which is little endian. However, I think that we should remove `ByteBufferRow` altogether and parse it manually into a `BinaryRowImpl`, so I'll rework this logic and will bring `putShort` back